components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/webextension/GeckoWebExtension.kt

```diff
@@ -142,6 +151,64 @@ class GeckoWebExtension(
             connectedPorts.remove(portId)
         }
     }
+
+    /**
+     * See [WebExtension.registerActionHandler].
+     */
+    override fun registerActionHandler(actionHandler: ActionHandler) {
+        if (!supportActions) {
+            logger.error("Attempt to register default action handler but browser and page " +
+                "action support is turned off for this extension: $id")
+            return
+        }
+
+        val actionDelegate = object : GeckoNativeWebExtension.ActionDelegate {
+
+            override fun onBrowserAction(
+                ext: GeckoNativeWebExtension,
+                // Session will always be null here for the global default delegate
+                session: GeckoSession?,
+                action: GeckoNativeWebExtensionAction
+            ) {
+                actionHandler.onBrowserAction(this@GeckoWebExtension, null, action.toBrowserAction())
+            }
+        }
+
+        nativeExtension.setActionDelegate(actionDelegate)
+    }
+
+    /**
+     * See [WebExtension.registerActionHandler].
+     */
+    override fun registerActionHandler(session: EngineSession, actionHandler: ActionHandler) {
+        if (!supportActions) {
+            logger.error("Attempt to register action handler on session but browser and page " +
```

Comment:
Nice 👍

Reply:
Hehe, yers, I forgot that before :). I just want to turn this off by default for built-in extensions, we can turn it on for third-party extensions, once we get the different methods.